packages/react-mutation-mapper/src/component/column/Signal.tsx

```diff
@@ -46,12 +49,48 @@ export function getSignalData(
         variantAnnotation.signalAnnotation.annotation &&
         variantAnnotation.signalAnnotation.annotation.length > 0
     ) {
-        variantAnnotation.signalAnnotation.annotation.forEach(annotation => {
-            // only have one germline annotation
-            if (annotation.mutationStatus.includes('germline')) {
-                signalData = extendMutations([annotation])[0];
+        // if mutation is somatic OR germline, get annotation depands on mutationType
+        if (variantAnnotation.signalAnnotation.annotation.length === 1) {
+            if (
+                (mutationType === Pathogenicity.BIALLELIC &&
+                    variantAnnotation.signalAnnotation.annotation[0].mutationStatus.includes(
+                        'germline'
+                    )) ||
+                (mutationType === Pathogenicity.SOMATIC &&
+                    variantAnnotation.signalAnnotation.annotation[0].mutationStatus.includes(
+                        'somatic'
+                    ))
+            ) {
+                signalData = extendMutations([
+                    variantAnnotation.signalAnnotation.annotation[0],
+                ]);
             }
-        });
+        }
+        // if mutation is both somatic AND germline, get annotation for both or depands on mutationType
+        else {
+            // if mutationType is undefined, get annotation for both somatic and germline
```

Comment:
are we getting annotation for both somatic and germline here? or actually in the `else` block?

Reply:
@onursumer If no `mutationType` is defined then get both, if `mutationType` is defined then we get one of them based on `mutationType`. 
Actually we don't have a use case for getting both germline and somatic for now, because in signal column of mutations table we only show germline frequencies, but in the future we probably want to change it to show both somatic and germline, or add a new column to show somatic, so I just leave it a possibility to be able to get both